apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/_components/DocumentEditor/Editor/Playground/DocumentParams/HistoryLogParams/useLogHistoryParams.ts

```diff
@@ -0,0 +1,151 @@
+    return paramValue
+      ? JSON.stringify(paramValue)
+      : typeof inputValue === 'string'
+        ? inputValue
+        : inputValue.value
+  } catch {
+    return typeof inputValue === 'string' ? inputValue : inputValue.value
+  }
+}
+function mapLogParametersToInputs({
+  inputs,
+  parameters,
+}: {
+  inputs: PlaygroundInputs
+  parameters: DocumentLog['parameters'] | undefined
+}): PlaygroundInputs | undefined {
+  const params = parameters ?? {}
+  // No parameters
+  if (!Object.keys(params).length) return undefined
+
+  return Object.entries(inputs).reduce((acc, [key, inputValue]) => {
+    const includedInPrompt = !!params[key]
+    acc[key] = {
+      value: getValue({ paramValue: params[key], inputValue }),
+      includedInPrompt,
+    }
+    return acc
+  }, {} as PlaygroundInputs)
+}
+
+const ONLY_ONE_PAGE = '1'
+export function useLogHistoryParams({
+  inputs,
+  setInputs,
+  setForceReadMetadata,
+}: {
+  inputs: PlaygroundInputs
+  setInputs: (newInputs: PlaygroundInputs) => void
+  setForceReadMetadata: ReactStateDispatch<boolean>
+}) {
+  const mounted = useRef(false)
+  const document = useCurrentDocument()
+  const { saveRowInfo, selectedRow } = useSelectedLogRow({ document })
+  const selectedLogUuid = selectedRow?.documentLogUuid
+  const { project } = useCurrentProject()
+  const { commit } = useCurrentCommit()
+  const [page, setPage] = useState<number>(1)
+  const { data: pagination, isLoading: isLoadingCounter } =
+    useDocumentLogsPagination({
+      projectId: project.id,
+      commitUuid: commit.uuid,
+      documentUuid: document.documentUuid,
+      page: page.toString(),
+      pageSize: ONLY_ONE_PAGE,
+    })
+  const [paginatedLog, setPaginatedLog] = useState<
+    DocumentLogWithMetadataAndError | undefined
+  >()
+  const onLogFetched = useCallback(
+    (logs: DocumentLogWithMetadataAndError[]) => {
+      const log = logs[0]
+      if (!log) return
+      setPaginatedLog(log)
+      saveRowInfo({ documentLogUuid: log.uuid })
+    },
+    [saveRowInfo, setPaginatedLog],
+  )
+  const { isLoading: isLoadingList } = useDocumentLogs({
+    // NOTE: Nasty hack to avoid fetching all the logs when
+    // we have a selected one log UUID in local storage.
+    //
+    // What we do here is set `documentUuid` to undefined
+    // so this way the fetcher in this hook will not be called
+    documentUuid: !selectedLogUuid ? document.documentUuid : undefined,
+    commitUuid: commit.uuid,
+    projectId: project.id,
+    page: String(page),
+    pageSize: ONLY_ONE_PAGE,
+    onFetched: onLogFetched,
+  })
+  const { isLoading: isLoadingLog } = useDocumentLogWithMetadata({
+    documentLogUuid: selectedLogUuid ?? paginatedLog?.uuid,
+    onFetched: (log) => {
+      const newInputs = mapLogParametersToInputs({
+        inputs,
+        parameters: log?.parameters,
+      })
+
+      if (!newInputs) return
+
+      setInputs(newInputs)
+
+      if (!mounted?.current) {
+        setForceReadMetadata(true)
```

Comment:
Metadata is read in a `useEffect` once when the editor loads. But the fresh data from the selected log comes after. So here what I do is to trigger a manual `readMetadata` by setting this flag to true

Reply:
hmm this is weird it looks like if reading the metadata should be in a use memo rather than a use effect and react to inputs changing

Reply:
inputs dependends on metadata changing. I can't make metadata dependent on inputs. It start sending read metadata to the worker non stop